lib/internal/loader/Loader.js

```diff
@@ -24,112 +27,89 @@ function getURLStringForCwd() {
   }
 }
 
-/* A Loader instance is used as the main entry point for loading ES modules.
- * Currently, this is a singleton -- there is only one used for loading
- * the main module and everything in its dependency graph. */
+
 class Loader {
   constructor(base = getURLStringForCwd()) {
-    if (typeof base !== 'string') {
+    if (typeof base !== 'string')
       throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'base', 'string');
-    }
 
-    this.moduleMap = new ModuleMap();
-    this.base = base;
     // The resolver has the signature
     //   (specifier : string, parentURL : string, defaultResolve)
     //       -> Promise<{ url : string,
-    //                    format: anything in Loader.validFormats }>
+    //                    translate: string }>
     // where defaultResolve is ModuleRequest.resolve (having the same
     // signature itself).
-    // If `.format` on the returned value is 'dynamic', .dynamicInstantiate
+    // If `.translate` on the returned value is 'dynamic', .dynamicTranslate
     // will be used as described below.
-    this.resolver = ModuleRequest.resolve;
-    // This hook is only called when resolve(...).format is 'dynamic' and has
+    this[Loader.resolve] = defaultResolve;
+    // This hook is only called when resolve(...).translate is 'dynamic' and has
     // the signature
     //   (url : string) -> Promise<{ exports: { ... }, execute: function }>
     // Where `exports` is an object whose property names define the exported
     // names of the generated module. `execute` is a function that receives
     // an object with the same keys as `exports`, whose values are get/set
     // functions for the actual exported values.
-    this.dynamicInstantiate = undefined;
-  }
+    this[kDynamicTranslate] = undefined;
 
-  hook({ resolve = ModuleRequest.resolve, dynamicInstantiate }) {
-    // Use .bind() to avoid giving access to the Loader instance when it is
-    // called as this.resolver(...);
-    this.resolver = resolve.bind(null);
-    this.dynamicInstantiate = dynamicInstantiate;
+    this.base = base;
+    this.registry = new Registry();
```

Comment:
Please use current standard terminology: https://html.spec.whatwg.org/multipage/webappapis.html#concept-settings-object-module-map

Reply:
I agree, the code should keep using that terminology. It sucks that that really makes it less comprehensible, though … so I would really not mind any kind of comments to explain what’s happening